lib/ibm/cloud/sdk/vpc/helpers/response.rb

```diff
@@ -0,0 +1,73 @@
+# typed: true
+# frozen_string_literal: true
+
+require 'http'
+
+module RestClient
+  # Update restclient reponse with additional methods.
+  class Response
+    def array
+      r = json
+      return r if r.instance_of?(Array)
+
+      raise "Expected array in reponse not #{r.class}."
+    end
+
+    def hash
+      r = json
+      return r if r.instance_of?(Hash)
+
+      raise "Expected hash in reponse not #{r.class}."
+    end
+
+    def subkey(key)
+      r = hash
+      return r.fetch(key.to_sym) if r.instance_of?(Hash) && r.key?(key)
+
+      raise "Key not found in #{r}."
+    end
+
+    def json
+      begin
+        r = JSON.parse(body)
+      rescue StandardError
+        raise "Error while parsing response body. #{body}"
+      end
+      r
+    end
+  end
+end
+
+module HTTP
+  # A HTTP response.
+  class Response
```

Comment:
We cannot patch core classes in a gem

Reply:
I'll have to rework my approach then. I guess each request is wrapped in a response with a response component and a json method. I don't want to throw away the response object.

Reply:
I've removed this and added a wrapped Response object in the connection class.